src/negative_tests.rs

```diff
@@ -472,6 +472,29 @@ fn test_generic_custom_type_mismatched() {
     ));
 }
 
+#[test]
+fn test_generic_mutated_cst_var_in_loop() {
+    let code = r#"
+        fn gen(const LEN: Field) -> [Field; LEN] {
+            return [0; LEN];
+        }
+        
+        fn main(pub xx: Field) {
+            let mut loopvar = 1;
+            for ii in 0..3 {
+                loopvar = loopvar + 1;
+            }
+            let arr = gen(loopvar);
```

Comment:
I'm not sure I understand why this code shouldn't compile

Reply:
Because we don't support for loop unrolling, so the `loopvar` won't end up with the correct constant value during MAST. 

With the wrong constant value for the `loopvar`, the generic function will be instantiated with wrong generic parameter.

Reply:
what would it take to make this code work though? I feel like we should support simple constant folding like this

Reply:
as long as the `gen(loopvar)` is not accepting the `loopvar` as its const generic parameter, it would work.